lib/matplotlib/colors.py

```diff
@@ -506,6 +506,11 @@ def to_rgba_array(c, alpha=None):
 
     if alpha is not None:
         rgba[:, 3] = alpha
+        if isinstance(c, Sequence):
+            # ensure that an explicit alpha does not overwrite full transparency
+            # for "none"
+            none_mask = np.array([isinstance(cc, str) and cc == "none" for cc in c])
```

Comment:
I don't think this needs to be an array, does it?
```suggestion
            none_mask = [cbook._str_equal(cc, "none") for cc in c]
```

Reply:
Yes. I took the liberty to commit. Please squash-merge.